src/app/utils/query-url-sanitization.ts

```diff
@@ -1,67 +1,99 @@
+import { GRAPH_URL } from '../services/graph-constants';
 import { parseSampleUrl } from './sample-url-generation';
 
-// Matches patterns like "('<1f7ff346-c174-45e5-af38-294e51d9969a>')" or "('<key>')"
-const keyIdRegex = /\(\'\<?\{?[ ?0-9a-zA-Z-]*\}?\>?\'\)/g;
-
-// Matches patterns like "(query=<key>)" or "(itemat=<key>,mode=<mode>)"
-const functionParamInitialRegex = /(?<=\=).*?(?=,)/g;
-const functionParamFinalRegex = /(?<=\=).[^,]*(?=\))/g;
-
 // Matches pattterns within quotes e.g "displayName: Gupta"
 const quotedTextRegex = /"([^"]*)"/g;
 
-// Matches PII patterns
-const numberRegex =  /(?<=\/)\d+\b/g; // number between forward slashes
-const emailRegex = /([a-z0-9_\-.+]+)@\w+(\.\w+)*/gi;
-const guidRegex =
- /(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}/gi;
-
-
 export function sanitizeQueryUrl(url: string): string {
   url = decodeURIComponent(url);
 
   // Extract query string
-  const { search } = parseSampleUrl(url);
+  const { search, queryVersion } = parseSampleUrl(url);
   const queryString: string = sanitizeQueryParameters(search);
 
   // Drop query string
   let sanitizedUrl = url.split('?')[0];
 
-  // Normalize parameters in param=<arbitraryKey> format with param={value} format
-  sanitizedUrl = sanitizedUrl.replace(functionParamInitialRegex, '<value>');
-  sanitizedUrl = sanitizedUrl.replace(functionParamFinalRegex, '<value>');
+  /*
+  * Remove url prefix and query version.
+  * They have special characters used in the check that follows
+  */
+  const urlPrefix = `${GRAPH_URL}/${queryVersion}`;
+  sanitizedUrl = sanitizedUrl.replace(urlPrefix, '');
 
-  // Replace IDs and ID placeholders with generic {id}
-  sanitizedUrl = sanitizedUrl.replace(keyIdRegex, '/<id>');
-  sanitizedUrl = sanitizedUrl.replace(guidRegex, '<guid>');
-  sanitizedUrl = sanitizedUrl.replace(numberRegex, '<number>');
+  const sections = sanitizedUrl.split('/');
+  sections.forEach(sec => {
+    if (containsIdentifier(sec)) {
```

Comment:
It won't always be true since we don't follow RFC 3987, which we couldn't since all segments could be an identifier. `containsIdentifier()` works on serviceable identification of identifiers, but should be considered incomplete. We will want a work item to track URL key sanitization based on entity collections. 

This comment is for information only and can be resolved.